src/libguac/client.c

```diff
@@ -544,12 +273,12 @@ int guac_client_add_user(guac_client* client, guac_user* user, int argc, char**
 
     int retval = 0;
 
-    pthread_rwlock_wrlock(&(client->__users_lock));
-
     /* Call handler, if defined */
     if (client->join_handler)
         retval = client->join_handler(user, argc, argv);
 
+    pthread_rwlock_wrlock(&(client->__users_lock));
```

Comment:
So that implementations of the `join_handler` and the `leave_handler` need not be aware of the internals of `guac_client` (that the `guac_client` will internally acquire a lock on the user list, preventing use of any broadcast socket).

Prior to this, writing to a broadcast socket within join/leave lead to undefined results (most likely a segfault). Now, writing to the broadcast socket within join/leave sends data to all users except the one joining/leaving.